### PR TITLE
[fix bug 1262189] Raise 404 on all exceptions in list_reports.

### PR DESCRIPTION
**Pull request by akatsoulas**
@johngian quick r?

### CODE REVIEW

**Comment by johngian** [comment]
I suggest we just extend the exceptions to also capture: ``MultiValueDictKeyError``

**Comment by akatsoulas** [comment]
Nope because if you try to change the query, a different one will be raised. And truth be told if we don't get exactly what we are aiming for, we should raise 404

**Comment by johngian** [comment]
r+

File: remo/reports/views.py [hunk 1]
```diff
@@ -200,14 +200,15 @@ def list_ng_reports(request, mentor=None, rep=None, functional_area_slug=None):
...
+    year = request.GET.get('year')
+    if month and year:
         try:
-            month = month2number(request.GET['month'])
+            month = month2number(month)
             # Make sure that year is an integer too
-            int(request.GET['year'])
-        except (TypeError, ValueError):
+            year = int(year)
+        except(TypeError, ValueError):
```
Comment (johngian) [inline remo/reports/views.py]: Please remove ``(``, ``)``

**Comment by johngian** [comment]
r+wc

**Pull request merged**